File: backend/app/schemas/voice.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class TranscribeResponse(BaseModel):
    success: bool
    text: Optional[str] = None
    original_transcript: Optional[str] = None
    normalized_transcript: Optional[str] = None
    detected_language: Optional[str] = None
    confidence: Optional[float] = None
    low_confidence: bool = False
    request_id: Optional[str] = None
    latency_ms: Optional[int] = None
    stt_model: Optional[str] = None
    error: Optional[str] = None
    message: Optional[str] = None


class SynthesizeRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000)
    response_language: str = Field(..., min_length=2, max_length=8)
    request_id: Optional[str] = None


class VoiceCitizenMessage(BaseModel):
    """Localized citizen-facing voice status/error strings."""

    key: str
    language: str
    message: str

File: mobile/src/types/voice.ts
/** Voice API types matching backend/app/schemas/voice.py */

export interface TranscribeResponse {
  success: boolean;
  text?: string | null;
  original_transcript?: string | null;
  normalized_transcript?: string | null;
  detected_language?: string | null;
  confidence?: number | null;
  low_confidence?: boolean;
  request_id?: string | null;
  latency_ms?: number | null;
  stt_model?: string | null;
  error?: string | null;
  message?: string | null;
}

export interface SynthesizeRequest {
  text: string;
  response_language: string;
  request_id?: string | null;
}

export type VoicePermissionState = 'unknown' | 'granted' | 'denied' | 'blocked';

export type VoiceInputPhase =
  | 'idle'
  | 'requesting_permission'
  | 'recording'
  | 'stopping'
  | 'uploading'
  | 'transcribing'
  | 'transcribed'
  | 'error';

export type TtsPlaybackPhase = 'idle' | 'loading' | 'playing' | 'error';

export interface VoiceSendOptions {
  input_mode: 'voice';
  stt_language: string;
  voice_request_id?: string;
}
